module/data/actor/fields/travel-field.mjs

```diff
@@ -0,0 +1,133 @@
+import { convertLength, convertTravelSpeed, simplifyBonus } from "../../../utils.mjs";
+import FormulaField from "../../fields/formula-field.mjs";
+import MappingField from "../../fields/mapping-field.mjs";
+
+const { StringField } = foundry.data.fields;
+
+/**
+ * @typedef {"slow"|"normal"|"fast"} TravelPace5e
+ */
+
+/**
+ * @typedef {object} TravelData
+ * @property {TravelPace5e} [pace]            Current travel pace.
+ * @property {Record<string, string>} speeds  Formulas for various travel speeds.
+ * @property {string} units                   Movement used to measure the various travel speeds.
+ */
+
+/**
+ * Field for storing travel data.
+ */
+export default class TravelField extends foundry.data.fields.SchemaField {
+  constructor(fields={}, { initialUnits=null, ...options }={}) {
+    const numberConfig = { required: true, nullable: true, min: 0, step: 0.1, initial: null };
+    fields = {
+      pace: new StringField({
+        required: true, blank: false, initial: "normal", choices: () => CONFIG.DND5E.travelPace,
+        label: "DND5E.Travel.Label"
+      }),
+      speeds: new MappingField(new FormulaField({ deterministic: true }), {
+        initialKeys: CONFIG.DND5E.travelTypes, initialKeysOnly: true
+      }),
+      units: new StringField({
+        required: true, nullable: true, blank: false, initial: initialUnits, label: "DND5E.UNITS.TRAVEL.Label"
+      }),
+      ...fields
+    };
+    Object.entries(fields).forEach(([k, v]) => !v ? delete fields[k] : null);
+    super(fields, options);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Number of hours of travel per day.
+   * @type {number}
+   */
+  static #HOURS_PER_DAY = 8; // TODO: Allow this to be configured
+
+  /* -------------------------------------------- */
+  /*  Data Preparation                            */
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare travel data.
+   * @this {ActorDataModel}
+   * @param {object} rollData  Actor's roll data.
+   */
+  static prepareData(rollData) {
+    const { movement, travel } = this.attributes;
+    const { pace, units } = travel;
+    travel.paces = {};
+    travel.max = 0;
+
+    const noMovement = this.parent.hasConditionEffect("noMovement");
+    const halfMovement = this.parent.hasConditionEffect("halfMovement");
+    const paceConfig = CONFIG.DND5E.travelPace[pace];
+    for ( const type of Object.keys(travel.speeds) ) {
+      let speed = Math.max(0, simplifyBonus(travel.speeds[type], rollData));
+      if ( noMovement ) speed = 0;
+      else if ( halfMovement ) speed *= 0.5;
+      travel.speeds[type] = speed;
+      if ( speed > travel.max ) travel.max = speed;
+      if ( speed ) travel.paces[type] = TravelField.convertSpeedToPace(speed, pace);
```

Comment:
Not passing units here means we can't apply the weird logic for PHB 'standard' travel pace for groups.